v2/instructor-client/src/EditCourse.js

```diff
@@ -0,0 +1,140 @@
+import React, { useState, useRef } from 'react';
+import {
+	MenuItem,
+	Select,
+	TextField,
+	Button,
+	Dialog,
+	Table,
+	TableBody,
+	TableContainer,
+	TableHead,
+	TableRow,
+	TableCell,
+} from '@material-ui/core';
+import axios from 'axios';
+import { serverURL } from './config';
+
+export default function EditCourse({ show, showOrHide, cid, course, fetchCourse }) {
+	const [settings, setSettings] = useState({ drawable: course.drawable, anonymity: course.anonymity });
+	const resultRef = useRef(null);
+	const nameRef = useRef(null);
+
+	/**
+	 * Change anonymity level:
+	 *   A: anonymous
+	 *   B: login required anonymous
+	 *   C: non-anonymous
+	 *   D: anonymous chat to classmates but not instructors
+	 */
+	const changeAnonymity = (e) => {
+		setSettings({ ...settings, anonymity: e.target.value });
```

Comment:
using multiple states might be more efficient than combining them into one object